airflow/providers/google/cloud/hooks/dataflow.py

```diff
@@ -680,22 +582,36 @@ def start_java_dataflow(
         :param location: Job location.
         :type location: str
         """
-        name = self._build_dataflow_job_name(job_name, append_job_name)
+        warnings.warn(
+            """"This method is deprecated.
```

Comment:
Are the new versions fully backward compatible with these ones?

Reply:
Yes. If someone uses this hook method the only difference should be this deprecation warning.